dio/lib/src/failed_request_interceptor.dart

```diff
@@ -9,12 +9,12 @@ class FailedRequestInterceptor extends Interceptor {
   final Hub _hub;
 
   @override
-  void onError(
+  Future<void> onError(
     DioError err,
     ErrorInterceptorHandler handler,
-  ) {
+  ) async {
     _hub.getSpan()?.throwable = err;
-    _hub.captureException(err);
+    await _hub.captureException(err);
```

Comment:
@ueman what do you think about this change?
That was the only integration not awaiting for the capture call

Reply:
Hm, not sure. I could imagine that the overrides could make some problems.

Reply:
I don't think this will cause any trouble.
Since DIO is waiting for `handler.next` being called, it doesn't matter if this function is synchronous or not.

Reply:
Yeah, that's not my worry. It's that we're changing the method signature from the overriden method. I would keep all the async in the method but keep just `void` as return type, so that it still matches the super method.

I'm also pretty sure that this is not the culprit for https://github.com/getsentry/sentry-dart/issues/500 since https://github.com/getsentry/sentry-dart/issues/500 is older than the Dio integration.

But I'm happy if it fixes the errors 🤷

Reply:
It's not since this error exists before the `dio` integration, but I'm looking for all possibilities of improvements for async stuff.

Reply:
It works btw, didn't have any issues running the code that is now async.